modules/040-node-manager/templates/capi-controller-manager/static-cluster.yaml

```diff
@@ -0,0 +1,39 @@
+{{- $prefix := "static" }}
+{{- $provider := "static" }}
+---
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: {{ $prefix }}
+  namespace: d8-cloud-instance-manager
+  {{- include "helm_lib_module_labels" (list . (dict "app" "capi-controller-manager")) | nindent 2 }}
+  annotations:
+    cluster.x-k8s.io/paused: "true"
+# A finalizer has been added to prevent cascading removing of all capi resources after deleting the cluster resource.
+  finalizers:
+    - deckhouse.io/capi-controller-manager
+spec:
+  clusterNetwork:
+    pods:
+      cidrBlocks:
+        - {{ .Values.global.clusterConfiguration.podSubnetCIDR }}
+    services:
+      cidrBlocks:
+        - {{ .Values.global.clusterConfiguration.serviceSubnetCIDR }}
+    serviceDomain: {{ .Values.global.clusterConfiguration.clusterDomain }}
+#  infrastructureRef:
+#    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha1
+#    kind: StaticCluster
+#    namespace: d8-cloud-instance-manager
+#    name: {{ $prefix }}
+---
+# empty kubeconfig secret for cluster
+apiVersion: v1
+data:
+  value: Cg==
```

Comment:
Is it not mistake?

Reply:
no, it is not a mistake
as described [here](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/book/src/developer/architecture/controllers/cluster.md#secrets) if we do not use kubeadm bootstrap provider, we should generate own cert authority and store it in the secret.
But we do not generate anymore for the cluster, because capi cluster is fake and needed only for controller work. So we generate secret with empty kubeconfig.
I added comment about it.